### PR TITLE
rename DownloadGitHubReleases task

### PR DESCRIPTION
**Pull request by gykuma**
- rename _DownloadGitHubRelease**s**_ task to *DownloadGitHubRelease*

- make **defaultVersionType** input uneditable

### CODE REVIEW

File: Tasks/DownloadBuildArtifactsV0/package-lock.json [hunk 1]
```diff
@@ -9,9 +9,9 @@
       "resolved": "https://registry.npmjs.org/align-text/-/align-text-0.1.4.tgz",
       "integrity": "sha1-DNkKVhCT810KmSVsIrcGlDP60Rc=",
       "requires": {
-        "kind-of": "3.2.2",
-        "longest": "1.0.1",
-        "repeat-string": "1.6.1"
+        "kind-of": "^3.0.2",
```
Comment (Lovakumar) [inline Tasks/DownloadBuildArtifactsV0/package-lock.json]: Undo changes in this file .. these are unrelated to DownloadGitHubRelease task

File: Tasks/DownloadGitHubReleaseV0/task.loc.json [hunk 2]
```diff
@@ -47,9 +47,6 @@
         "specificVersion": "Specific Version",
         "specificTag": "Specific Tag"
       },
-      "properties": {
```
Comment (Lovakumar) [inline Tasks/DownloadGitHubReleaseV0/task.loc.json]: Why are we removing editable?

File: make-options.json [hunk 3]
```diff
@@ -55,7 +55,7 @@
         "DotNetCoreInstallerV0",
         "DownloadBuildArtifactsV0",
         "DownloadFileshareArtifactsV0",
-        "DownloadGitHubReleasesV0",
+        "DownloadGitHubReleaseV0",
         "DownloadPackageV0",
```
Comment (Lovakumar) [inline make-options.json]: After completing this PR, please create corresponding PR in AzureDevOps repo.

File: Tasks/DownloadGitHubReleaseV0/main.ts [hunk 4]
```diff
@@ -10,8 +10,8 @@ var packagejson = require('./package.json');
 tl.setResourcePath(path.join(__dirname, 'task.json'));
 
 var taskJson = require('./task.json');
-const area: string = 'DownloadGitHubReleases';
-const userAgent: string = 'download-github-releases-task-' + packagejson.version;
+const area: string = 'DownloadGitHubRelease';
```
[the remark this answers is not in the document]
Comment (gykuma) [reply Tasks/DownloadGitHubReleaseV0/main.ts]: this is additional change to make names more consistent

File: Tasks/DownloadGitHubReleaseV0/task.loc.json (hunk 2 above)
Comment (gykuma) [reply Tasks/DownloadGitHubReleaseV0/task.loc.json]: input for this field has to be strictly from values in **options**

File: Tasks/DownloadBuildArtifactsV0/package-lock.json (hunk 1 above)
Comment (gykuma) [reply Tasks/DownloadBuildArtifactsV0/package-lock.json]: removed

File: make-options.json (hunk 3 above)
Comment (gykuma) [reply make-options.json]: sure.
